Connectors/ArcGIS/Speckle.Connectors.ArcGIS3/HostApp/ArcGISColorManager.cs

```diff
@@ -43,6 +48,151 @@ public List<ColorProxy> UnpackColors(List<(MapMember, int)> mapMembersWithDispla
     return ColorProxies.Values.ToList();
   }
 
+  /// <summary>
+  /// Parse Color Proxies and stores in ObjectColorsIdMap the relationship between object ids and colors
+  /// </summary>
+  /// <param name="colorProxies"></param>
+  /// <param name="onOperationProgressed"></param>
+  public void ParseColors(List<ColorProxy> colorProxies, Action<string, double?>? onOperationProgressed)
+  {
+    // injected as Singleton, so we need to clean existing proxies first
+    ObjectColorsIdMap = new();
+    var count = 0;
+    foreach (ColorProxy colorProxy in colorProxies)
+    {
+      onOperationProgressed?.Invoke("Converting colors", (double)++count / colorProxies.Count);
+      foreach (string objectId in colorProxy.objects)
+      {
+        Color convertedColor = Color.FromArgb(colorProxy.value);
+        ObjectColorsIdMap.TryAdd(objectId, convertedColor);
+      }
+    }
+  }
+
+  /// <summary>
+  /// Create a new CIMUniqueValueClass for UniqueRenderer per each object ID
+  /// </summary>
+  /// <param name="tc"></param>
+  /// <param name="speckleGeometryType"></param>
+  private CIMUniqueValueClass CreateColorCategory(TraversalContext tc, esriGeometryType speckleGeometryType)
+  {
+    Base baseObj = tc.Current;
+
+    // declare default white color
+    Color color = Color.FromArgb(255, 255, 255, 255);
+
+    // get color moving upwards from the object
+    foreach (var parent in tc.GetAscendants())
+    {
+      if (parent.applicationId is string appId && ObjectColorsIdMap.TryGetValue(appId, out Color objColor))
+      {
+        color = objColor;
+        // POC: should we get the color of the closest parent and break, or walk up to the highest possible?
+        break;
+      }
```

Comment:
Should we get the color of the closest parent, or the highest in the hierarchy?

Reply:
Not sure I understand the question here: if the atomic object doesn't have a color, then the color of the closest parent with a color should be used

Reply:
Let's resolve this POC comment before merge